### PR TITLE
Cart REST API Doc was added

### PR DESCRIPTION
**Pull request by Sunagatov**

### CODE REVIEW

File: docs/CartRestApiDoc.md [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+{
+    "message": "Invalid 'itemList' value.",
+    "timestamp": "2023-06-04 18:24:54",
+    "code": 15
+}
+```
+
+--------------------------------------------------------------------------
+
+## 4. Update the amount of the specified item products in the cart
```
Comment (yevr19) [inline docs/CartRestApiDoc.md:365]: please use HTTP PUT method for it.

if you update the quantity of the specified product in the cart, correct url might look as follows:

HTTP PUT .../api/v1/cart/items/{item_id}/amount

File: docs/CartRestApiDoc.md [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+
+| Parameter         | In | Type  | Required | Description                                            |
+|-------------------|----|-------|----------|--------------------------------------------------------|
+| shoppingSessionId | body | string | true | The identifier of the cart                            |
+| itemId            | body | string | true | The identifier of the item                            |
+| change            | body | string | true | The amount of the specified item products in the cart |
+
+**Request Example:**
+
+POST /cart/items/
```
Comment (yevr19) [inline docs/CartRestApiDoc.md:387]: incorrect url

File: docs/CartRestApiDoc.md [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+| timestamp | timestamp | true | Indicating the failure time.                        |
+| code | integer   | true | Error code.    |
+
+**Response Examples:**
+
+```json
+{
+    "message": "The 'shoppingSessionId' cannot be empty.",
+    "timestamp": "2023-06-04 18:24:54",
+    "code": 12
```
[the remark this answers is not in the document]
Comment (Reyzis2021) [reply docs/CartRestApiDoc.md:200]: Кстати если мы будем использовать возможности спрингового error handler, для заполнения error response, то все значения мы можем задавать прямо там, и там же можно выбрать какой error code будем использовать , без того что бы харкодить его значение

**Review by Reyzis2021 — APPROVED** [review]

File: docs/CartRestApiDoc.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Sunagatov) [reply docs/CartRestApiDoc.md:200]: > please elaborate what '12' is meaning here? If we're using code as HTTP code - this sample is supposed to be illegal

I added `//TODO` comment as I haven't added any error codes to the project yet.

File: docs/CartRestApiDoc.md [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+}
+```
+
+```json
+{
+    "message": "Invalid 'shoppingSessionId' value.",
+    "timestamp": "2023-06-04 18:24:54",
+    "code": 14
```
[the remark this answers is not in the document]
Comment (Sunagatov) [reply docs/CartRestApiDoc.md:208]: I added `//TODO` comment as I haven't added any error codes to the project yet.

File: docs/CartRestApiDoc.md [hunk 5]
```diff
@@ -0,0 +1,515 @@
...
+### Request:
+
+| Parameter         | In | Type      | Required | Description                 |
+|-------------------|----|-----------|----------|-----------------------------|
+| shoppingSessionId | body | string    | true | The identifier of the cart |
+| itemList          | body | item list | true | The item identifiers list  |
+
+**Request Example:**
+
+POST /cart/items/
```
[the remark this answers is not in the document]
Comment (Sunagatov) [reply docs/CartRestApiDoc.md:251]: Fixed

File: docs/CartRestApiDoc.md (hunk 1 above)
Comment (Sunagatov) [reply docs/CartRestApiDoc.md:365]: Fixed

File: docs/CartRestApiDoc.md (hunk 2 above)
Comment (Sunagatov) [reply docs/CartRestApiDoc.md:387]: Fixed

File: docs/CartRestApiDoc.md [hunk 6]
```diff
@@ -0,0 +1,515 @@
...
+|--------------------|-----------|----------|-------------------------------|
+| itemId             | integer   | true | The identifier of the item    |
+| productsAmount     | integer   | true | The amount of products      |
+| totalProductsPrice | integer   | true | The total products price     |
+| product            | integer   | true | The product description       |
+
+**Response Example:**
+
+```json
+{
```
[the remark this answers is not in the document]
Comment (Sunagatov) [reply docs/CartRestApiDoc.md:293]: Not clear. Explain please

File: docs/CartRestApiDoc.md [hunk 7]
```diff
@@ -0,0 +1,515 @@
...
+    "timestamp": "2023-06-04 18:24:54",
+    "code": 16
+}
+```
+
+------------------------------------------------------------------------------
+
+## 3. Remove the item in the cart
+
+### HTTP DELETE .../api/v1/cart/items/
```
[the remark this answers is not in the document]
Comment (Sunagatov) [reply docs/CartRestApiDoc.md]: Okay, which alternatives you can offer?

File: docs/CartRestApiDoc.md [hunk 8]
```diff
@@ -0,0 +1,514 @@
...
+    "timestamp": "2023-06-04 18:24:54",
+    "code": 16 // TODO Create errorCode hierachy
+}
+```
+
+------------------------------------------------------------------------------
+
+## 3. Remove the item in the cart
+
+### HTTP DELETE .../api/v1/cart/{shoppingSessionId}/items/
```
Comment (UladzislauM) [inline docs/CartRestApiDoc.md:232]: I would add .../api/v1/cart/{shoppingSessionId}/items/{shoppingSessionItemId}.
If we talking about "THE item".

**Pull request merged**